docs/LoaderApplicationInterface.md

```diff
@@ -0,0 +1,663 @@
+> This lead to the concept of "instance layers" and "device layers".
+> It was decided by Khronos to deprecate the "device layer" functionality and
+> only consider "instance layers".
+> Therefore, `vkCreateDevice` will use the layers specified at
+`vkCreateInstance`.
+> Because of this, the following items have been deprecated:
+> * `VkDeviceCreateInfo` fields:
+>   * `ppEnabledLayerNames`
+>   * `enabledLayerCount`
+> * The `vkEnumerateDeviceLayerProperties` function
+
+
+### Meta-Layers
+
+Meta-layers are layers which really just contain an ordered list of other
+layers.
+This is to allow grouping layers together in a specified order so that they can
+interact properly.
+Originally, this was designed to group together the individual Vulkan
+Validation layers in the proper order to avoid conflicts.
+While the Vulkan Validation layer no longer uses this, VkConfig does use this
+to group layers together based on user's desires.
+More can be found out about this functionality through both the
+[VkConfig documentation](https://github.com/LunarG/VulkanTools/blob/master/vkconfig/README.md)
+and the section later on the [Override Layer](#override-layer).
+
+Meta-layers are detailed more in the
+[Meta-Layers](LoaderLayerInterface.md#meta-layers) section of the
+[LoaderLayerInterface.md](LoaderLayerInterface.md) file in this folder.
+
+
+### Implicit vs Explicit Layers
+
+![Different Types of Layers](./images/loader_layer_order.png)
+
+Explicit layers are layers which are enabled by an application (e.g. with the
+vkCreateInstance function as mentioned previously), or by an environment
+variable.
+
+Implicit layers are those which are enabled by their existence.
+For example, certain application environments (e.g. Steam or an automotive
+infotainment system) may have layers which they always want enabled for all
+applications that they start.
+Other implicit layers may be for all applications started on a given system
+(e.g. layers that overlay frames-per-second).
+Implicit layers are enabled automatically, whereas explicit layers must be
+enabled explicitly.
+
+Implicit layers have an additional requirement over explicit layers in that
+they require being able to be disabled by an environmental variable.
+This is due to the fact that they are not visible to the application and could
+cause issues.
+A good principle to keep in mind would be to define both an enable and disable
+environment variable so the users can deterministically enable the
+functionality.
+On Desktop platforms (Windows, Linux, and macOS), these enable/disable settings
+are defined in the layer's JSON file.
+
+Discovery of system-installed implicit and explicit layers is described later
+in the [Layer discovery](LoaderLayerInterface#layer-discovery)
+section in the
+[LoaderLayerInterface.md document](LoaderLayerInterface.md) document.
+For now, simply know that what distinguishes a layer as implicit or explicit
+is dependent on the Operating system, as shown in the table below.
```

Comment:
"Distinguishes a layer as implicit or explicit is dependent on the OS" is a weird choice of words, because what distinguishes the two really depends on *where* the layers are located on an OS more than the OS itself. 
I do see how android doesn't allow implicit layers, so that's justifiable, but otherwise its not super clear.

Reply:
That is nasty.  Will update.

Reply:
Done, please look over new version.